controller/files.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Flow Playground
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controller
+
+import (
+	"github.com/dapperlabs/flow-playground-api/blockchain"
+	"github.com/dapperlabs/flow-playground-api/model"
+	"github.com/dapperlabs/flow-playground-api/storage"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+type Files struct {
+	store      storage.Store
+	blockchain *blockchain.Projects
+}
+
+func NewFiles(
+	store storage.Store,
+	blockchain *blockchain.Projects,
+) *Files {
+	return &Files{
+		store:      store,
+		blockchain: blockchain,
+	}
+}
+
+func (f *Files) CreateFile(projectID uuid.UUID, input model.NewFile, fileType model.FileType) (*model.File, error) {
+	file := model.File{
+		ID:        uuid.New(),
+		ProjectID: projectID,
+		Title:     input.Title,
+		Type:      fileType,
+		Script:    input.Script,
+	}
+
+	err := f.store.InsertCadenceFile(&file)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to store file")
+	}
+
+	return &file, nil
+}
+
+func (f *Files) UpdateFile(input model.UpdateFile) (*model.File, error) {
+	var file model.File
+	err := f.store.UpdateCadenceFile(input, &file)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to update cadence file")
+	}
+
+	return &file, nil
+}
+
+func (f *Files) DeleteFile(scriptID, projectID uuid.UUID) error {
+	err := f.store.DeleteCadenceFile(scriptID, projectID)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete cadence file")
+	}
+
+	return nil
+}
+
+func (f *Files) CreateScriptExecution(input model.NewScriptExecution) (*model.ScriptExecution, error) {
```

Comment:
I feel this should be moved back to the script controller. The same is true for transactions. And DeployContract could be in a contracts controller. I think that because transactions might have more methods like getting the result of transaction and it doesn't really have to do anything with the files.

Reply:
My thought was that contract deployments, script and transaction executions all occur based on the file / file type. I think it's more simple if we don't even have separate controllers for contracts, scripts, transactions and just have the file controller which deals with files based on their type.

CreateTransactionExecution already returns the execution which contains the result?